packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -339,7 +339,7 @@ function createOpenTagMarkup(
     }
     let markup = null;
     if (isCustomComponent(tagLowercase, props)) {
-      if (!RESERVED_PROPS.hasOwnProperty(propKey)) {
+      if (!RESERVED_PROPS[propKey]) {
```

Comment:
Oh right. I forgot how JS objects work.

I think I'm out of ideas for fixing #12368 here. I'll check on Closure side if it can be made to understand hasOwnProperty as dynamic access so it can keep the properties like with bracket access.

Reply:
Could you give me more details on your strategy in general? Is this the only issue blocking you? It's a bit hard to believe given in how many places we use dot-access.